src/commands/generate.ts

```diff
@@ -0,0 +1,161 @@
+
+    const isURL = /^https?:/;
+    if (!isURL.test(this._mapBaseUrlToFolder.url.toLowerCase())) {
+      throw new Error('Invalid --map-base-url flag. The mapping <url>:<folder> requires a valid http/https url and valid folder with delimiter `:`.');
+    }
+  }
+}
+
+export default class Generate extends Command {
+  static description = 'Generator is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input.';
+
+  static examples: Example[] = [
+    'asyncapi generate asyncapi.yaml @asyncapi/html-template'
+  ];
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+    'disable-hook': Flags.string({
+      char: 'd',
+      description: 'disable a specific hook type or hooks from a given hook type',
+      multiple: true
+    }),
+    install: Flags.boolean({
+      char: 'i',
+      default: false,
+      description: 'installs the template and its dependencies (defaults to false)'
+    }),
+    debug: Flags.boolean({
+      description: 'to enable specific errors in the console'
+    }),
+    'no-overwrite': Flags.string({
+      char: 'n',
+      multiple: true,
+      description: 'glob or path of the file(s) to skip when regenerating'
+    }),
+    output: Flags.string({
+      char: 'o',
+      description: 'directory where to put the generated files (defaults to current directory)'
+    }),
+    'force-write': Flags.boolean({
+      default: false,
+      description: 'force writing of the generated files to given directory even if it is a git repo with unstaged files or not empty dir (defaults to false)'
+    }),
+    'watch-tempalte': Flags.boolean({
+      description: 'watches the template directory and the AsyncAPI document, and re-generate the files when changes occur. Ignores the output directory.'
+    }),
+    param: Flags.string({
+      char: 'p',
+      description: 'additional param to pass to templates',
+      multiple: true
+    }),
+    'map-base-url': Flags.string({
+      description: ' maps all schema references from base url to local folder'
+    })
+  }
+
+  static args = [
+    { name: 'asyncapi', description: 'Local path or url pointing to AsyncAPI specification file', required: true },
+    { name: 'template', description: 'Name of the generator template like for example @asyncapi/html-template or https://github.com/asyncapi/html-template' }
+  ]
+
+  async run() {
+    const { args, flags } = await this.parse(Generate); // NOSONAR
+    const asyncapi = await load(args['asyncapi']);
+    const template = args['template'];
+
+    const flagParser = new GenerateFlagParser(flags['disable-hook'], flags['param'], flags['map-base-url'] as string);
+
+    const params = flagParser.params();
+    const disableHooks = flagParser.disableHooks();
+    const mapBaseURLToFolder = flagParser.mapBaseUrlToFolder();
+
+    const generator = new AsyncAPIGenerator(template, flags.output || path.resolve(os.tmpdir(), 'asyncapi-generator'), {
```

Comment:
I know that we use `path.resolve(os.tmpdir(), 'asyncapi-generator')` as fallback in current CLI, but I think that we should annotate `output` flag as required.

Reply:
As output is required then `path.resolve(os.tmpdir(), 'asyncapi-generator')` is not needed.